src/services/expenses.js

```diff
@@ -0,0 +1,75 @@
+'use strict';
+
+const { Expense } = require('../models/Expense.js');
+const { Op } = require('sequelize');
+const { dateNormalize } = require('../utils/dateNormalize.js');
+
+async function getAll() {
+  return Expense.findAll();
+}
+
+function getAllByQuery(query) {
+  const select = { ...query };
+  const hasDateSelect = select.hasOwnProperty('from')
+    || select.hasOwnProperty('to');
+
+  if (hasDateSelect) {
+    Object.entries(select).forEach(([key, value]) => {
+      switch (key) {
+        case 'from':
+          select.spentAt = {
+            ...select.spentAt,
+            [Op.gte]: dateNormalize(value),
+          };
+
+          delete select.from;
+          break;
+
+        case 'to':
+          select.spentAt = {
+            ...select.spentAt,
+            [Op.lte]: dateNormalize(value),
+          };
+
+          delete select.to;
+          break;
+
+        default:
+          break;
+      }
+    });
+  }
```

Comment:
you could just use 2 ifs instead of this forEach

```suggestion
function getAllByQuery(query) {
  const select = { ...query };
  
  if (select.hasOwnProperty('from')) {
    select.spentAt = {
      [Op.gte]: dateNormalize(select.from),
    };
    delete select.from;
  }

  if (select.hasOwnProperty('to')) {
    select.spentAt = {
      ...select.spentAt,
      [Op.lte]: dateNormalize(select.to),
    };
    delete select.to;
  }
```

Reply:
You are right